### DISCUSSION TITLE
Is the standard scheduler thread safe?

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#541] [General] [answered]
### Version: 3.0.0-alpha3
 

Is it safe to call `ScheduleJob`, `TriggerJob`, etc. on the `IScheduler`s returned from `StdSchedulerFactory` from multiple threads?

**Comment by lahma** [comment, marked as answer]
Should be, until proven otherwise. Both RAMJobStore and and AdoJobStore use locks when they access job/trigger storage.
